### ISSUE TITLE
Make tidy.py check for links to spec pages

### ISSUE THREAD

**Issue by jdm** [servo/servo#5730]
Following up on #5729, let's avoid adding any new spec links that might break in the future. We can use the regex there to check for any that might otherwise sneak in.

Code: python/tidy.py

**Comment by dhananjay92** [comment]
I am trying to do this one. Since, I don't know RegExps in Python that well, so it may take some time for me to figure everything out.

**Comment by dhananjay92** [comment]
Done. Please have a look at the [changes](https://github.com/dhananjay92/servo/commit/cb8bf8536bd157291ea3f994b859dda7bc5f9513).

Tested it by changing the links to point to specific page. It gives errors in such cases.

Please let me know if there's better way of doing the same. *I am asking this because I am not sure if my approach is best way to detect such links.*

Also I am thinking of refactoring the code so that `check_whitespace()` and `check_whatwg_url()` take line instead of whole content. That way, both don't have to break the content into lines and iterate over them individually. So something like this...
```
def check_whitespace_url(content):
  # Break content into lines
  for each line
      check_whitespace(line)
      check_whatwg_url(line)

def check_whitespace(line)
    # check whitespace for line
def check_whatwg_url(line)
    # check urls for line

```

Please let me know your thoughts on it.

**Comment by jdm** [comment]
Yes, that sounds like a sensible refactoring to me!

**Comment by Manishearth** [comment]
fixed in #5747
